### PR TITLE
Simplify keywords; add `open`; re-export symbols

### PR DESCRIPTION
**Pull request by rnewman**
You can see how much cleaner this makes the vocabulary tests and other end-to-end examples.

### CODE REVIEW

**Comment by rnewman** [comment]
I seem to recall that @kitcambridge's work around pairing connections and Conn is relevant to `open`; we should smush those things together at some point.

**Comment by rnewman** [comment]
Whoever's free, please take a look!

File: tests/vocabulary.rs [hunk 1]
```diff
@@ -117,14 +120,34 @@ fn test_real_world() {
...
+    // ::new is helpful. ::default is not.
+    assert!(vocabulary::AttributeBuilder::default()
+                  .value_type(ValueType::String)
+                  .multival(true)
+                  .fulltext(true)
+                  .build()
+                  .validate(|| "Foo".to_string())
+                  .is_err());
+
+    assert!(!vocabulary::AttributeBuilder::new()
```
Comment (fluffyemily) [inline tests/vocabulary.rs]: Could this be
```
assert!(vocabulary::AttributeBuilder::new()
    .value_type(ValueType::String)
    ...
    .is_ok());
```

instead? It's much clearer. I missed the negation at the beginning first time I read it.

Comment (rnewman) [reply tests/vocabulary.rs]: Yes! Excellent idea.

File: tools/cli/src/mentat_cli/store.rs [hunk 2]
```diff
@@ -8,66 +8,68 @@
...
 
 use mentat::query::QueryResults;
 
-use mentat::conn::Conn;
 use mentat_db::types::TxReport;
 
-pub struct Store {
-    handle: rusqlite::Connection,
-    conn: Conn,
+pub struct OpenStore {
```
[the remark this answers is not in the document]
Comment (rnewman) [reply tools/cli/src/mentat_cli/store.rs]: Done. I flattened it into `Repl`.

**Comment by rnewman** [comment]
2614f498beecd6744a9bf8bc47c539a8c85e998b

**Pull request closed without merging**